src/runtime/execution.rs

```diff
@@ -80,7 +77,7 @@ impl Execution {
             match state.current_task {
                 ScheduledTask::Some(tid) => {
                     let task = state.get(tid);
-                    Some((Rc::clone(&task.future), task.waker()))
+                    Some(Rc::clone(&task.inner))
```

Comment:
Re: Removing AtomicTask.
I'm running into an issue with lifetimes when I try to return the task here.
Specifically, if I want to return `Some(state.get(tid))`, I get the following complaint:
```
lifetime may not live long enough
returning this value requires that `'1` must outlive `'2`rustc
execution.rs(73, 44): has type `&'1 mut ExecutionState`
execution.rs(73, 49): return type of closure is Option<&'2 Task>
```
Is there a way to fix this without changing `ExecutionState`'s `tasks` in `Rc`s?

Reply:
Answer: I should be returning the continuation `Rc` here instead.